bin/ruby-build

```diff
@@ -1195,13 +1196,36 @@ usage() {
   [ -z "$1" ] || exit "$1"
 }
 
+# list all versions
 list_definitions() {
   { for DEFINITION_DIR in "${RUBY_BUILD_DEFINITIONS[@]}"; do
       [ -d "$DEFINITION_DIR" ] && ls "$DEFINITION_DIR"
     done
   } | sort_versions | uniq
 }
 
+# list only latest stable versions excluding RC, preview, dev and EoL'ed
+list_latest_definitions() {
+  { for DEFINITION_DIR in "${RUBY_BUILD_DEFINITIONS[@]}"; do
+      [ -d "$DEFINITION_DIR" ] && \
+      (
+        cd "$DEFINITION_DIR";
+        grep -L -e warn_eol * | \
+        grep -v -e '-rc[0-9]*$' -e '-preview[0-9]*$' -e '-dev$'
+      )
+    done
+  } | filter_previous_versions | sort_versions | uniq
+}
+
+filter_previous_versions() {
+  # sort in this function looks redundunt but it is necessary
+  # rbx-3.99 appears latest unless the sort
+  sed 'h; s/[+-]/./g; s/.p\([[:digit:]]\)/.z.\1/; s/$/.z/; G; s/\n/ /' | \
+    LC_ALL=C sort -t. -k 1,1 -k 2,2n -k 3,3n -k 4,4n -k 5,5n | \
```

Comment:
Indeed.

Reply:
Nope, actually we can't. In that function, intermediate output of is required (before `awk { print $2 }`). It cannot be simply replaced. Re-using common part both from `sort_versions` and `extract_latest_versions` is possible.

@mislav What do you think?

Reply:
It can be commonalized like this.

```shell
extract_latest_versions() {
  _some_briliant_name | \
    sed 's/[+.]/ /; s/[0-9].*z //; s/^\([0-9].[0-9]\)/mri\1 \1/' | \
    awk '{ latest[$1] =$2 } END{ for(key in latest) { print latest[key] } }'
}

sort_versions() {
    _some_briliant_name | awk '{print $2}'
}

_some_briliant_name() {
  sed 'h; s/[+-]/./g; s/.p\([[:digit:]]\)/.z.\1/; s/$/.z/; G; s/\n/ /' | \
    LC_ALL=C sort -t. -k 1,1 -k 2,2n -k 3,3n -k 4,4n -k 5,5n
}
```